packages/wallet/src/redux/reducer.ts

```diff
@@ -63,9 +65,23 @@ function routeToProtocolReducer(state, action) {
   }
 }
 
-function routeToNewProcessInitializer(state, action) {
-  // TODO: Call the process creator
-  return state;
+function routeToNewProcessInitializer(
+  state: states.Initialized,
+  action: actions.protocol.NewProcessAction,
+) {
+  switch (action.type) {
+    case actions.indirectFunding.FUNDING_REQUESTED:
+      const { protocolState, sharedData } = indirectFunding.initialize(
+        action,
+        states.sharedData(state),
+      );
+
+      return startProcess(state, sharedData, action, protocolState);
+    default:
+      return state;
+    // TODO: Why is the discriminated union not working here?
```

Comment:
Weird, what was the error from `unreachable` ?

Reply:
```
Argument of type '{ type: "WALLET.INDIRECT_FUNDING.FUNDING_REQUESTED"; channelId: string; playerIndex: PlayerIndex; protocol: WalletProtocol; }' is not assignable to parameter of type 'never'.
```

?

Reply:
I've had this problem when there's only one type before. I think if there's only one possibility for a type, typescript doesn't bother narrowing it in the switch statement.